c7n/resources/sagemaker.py

```diff
@@ -66,14 +68,107 @@ class resource_type(object):
         service = 'sagemaker'
         enum_spec = ('list_training_jobs', 'TrainingJobSummaries', None)
         detail_spec = (
-            'describe_training_job', 'TrainingJobName',
-            'TrainingJobName', None)
+            'describe_training_job', 'TrainingJobName', 'TrainingJobName', None)
         id = 'TrainingJobArn'
         name = 'TrainingJobName'
         date = 'CreationTime'
         dimension = None
         filter_name = None
 
+    permissions = (
+        'sagemaker:ListTrainingJobs', 'sagemaker:DescribeTrainingJobs',
+        'sagemaker:ListTags')
+
+    def __init__(self, ctx, data):
+        super(SagemakerJob, self).__init__(ctx, data)
+        self.queries = QueryFilter.parse(
+            self.data.get('query', [
+                {'StatusEquals': 'InProgress'}]))
+
+    def resources(self, query=None):
+        q = self.query_filter()
+        if q is not None:
+            query = query or {}
```

Comment:
seems a little odd wrt to shadowing/ignoring the function parameter `query`.. i think i had forgotten how akward the query filter pattern implementation was.

Reply:
since you're implementing `query_filter`, why would you not just have it with this formatting already?  There does not appear to be a reason to push this iteration outside of query_filter.